docs/spec_draft.md

```diff
@@ -2512,6 +2513,85 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.triangular_solve
+
+### Semantics
+
+Solves batches of systems of linear equations with lower or upper triangular
+coefficient matrices.
+
+More formally, given `a` and `b`, `result[i0, ..., iR-3, :, :]` is the solution
+to `op(a[i0, ..., iR-3, :, :]) * x = b[i0, ..., iR-3, :, :]` when `left_side` is
+`true` or `x * op(a[i0, ..., iR-3, :, :]) = b[i0, ..., iR-3, :, :]` when
+`left_side` is `false`, solving for the variable `x` where `op(a)` is determined
+by `transpose_a`, which can be one of the following:
+  * `NO_TRANSPOSE`: Perform operation using `a` as-is.
+  * `TRANSPOSE`: Perform operation on transpose of `a`.
+  * `ADJOINT`: Perform operation on conjugate transpose of `a`.
+
+Input data is read only from the lower triangle of `a`, if `lower` is `true` or
+upper triangle of `a`, otherwise. Output data is returned in the same triangle;
+the values in the other triangle are implementation-defined.
+
+If `unit_diagonal` is true, then the implementation can assume that the diagonal
+elements of `a` are equal to 1, otherwise the behavior is undefined.
+
+### Inputs
+
+| Name            | Type                                               |
+|-----------------|----------------------------------------------------|
+| `a`             | tensor of floating-point or complex type           |
+| `b`             | tensor of floating-point or complex type           |
+| `left_side`     | constant of type `i1`                              |
+| `lower`         | constant of type `i1`                              |
+| `unit_diagonal` | constant of type `i1`                              |
+| `transpose_a`   | enum of `NO_TRANSPOSE`, `TRANSPOSE`, and `ADJOINT` |
+
+### Outputs
+
+| Name     | Type                                     |
+|----------|------------------------------------------|
+| `result` | tensor of floating-point or complex type |
+
+### Constraints
+
+  * (C1) `a` and `b` have the same element type
+  * (C2) rank(`a`) $=$ rank(`b`) $\ge$ 2.
+  * (C3) dim(`a`, -2) = dim(`a`, -1).
+  * (C4) The relationship between `shape(a)` and `shape(b)` is as follows:
+    * For all `i` $\in$ [0, R-3], dim(`a`, `i`) $=$ dim(`b`, `i`).
+    * `dim(a, R-2)` = `dim(b, left_side ? R-2 : R-1)`.
```

Comment:
`dim(a, R-1)` = `dim(a, R-2)` = `dim(b, left_side ? R-2 : R-1)`.

@ghpvnist how about we remove C2 and the third sub point and combine them like this?